src/msw/webview_edge.cpp

```diff
@@ -25,10 +25,85 @@
 #include "wx/msw/private/cotaskmemptr.h"
 #include "wx/msw/private/webview_edge.h"
 
-#include <wrl/event.h>
-#include <Objbase.h>
+#include <objbase.h>
+#include <atomic>
```

Comment:
pre-C++11 doesn't support lambda, so obviously it won't work, but to be honest, people should just update their compilers if they are still using pre-C++11. C++11 isn't new now. There are always new stuff coming, e.g. C++17 CTAD. We shall use CTAD instead of op () to collect argument types of the lambda, but it will require C++17. The way this patch deals with lambda is actually risky, because it presumes lambda provides op () but lambda is implementation-dependent. The op () works anyway when using Visual C++, GNU C++ and Clang C++.

Reply:
Updating to a new gcc version is usually not a big deal, but switching from one MSVS version to another one can be non-obvious because they're not ABI-compatible, in general. So if this code previously compiled with older MSVS without C++14 support (which is still a question, I'm not sure if it did), I'd rather keep using `wrl/event.h` for it rather than breaking it.

As for CTAD and C++17, I'm not sure why would we need it when WRL does not.

Reply:
I'm really not sure if WRL requires C++14. But it might make sense that it does because I think it required VS 2015 which included C++14 (at least partly) if I remember correctly.